app/src/main/java/com/duckduckgo/app/browser/WebViewRequestInterceptor.kt

```diff
@@ -71,13 +75,28 @@ class WebViewRequestInterceptor(
 
         val url = request.url
 
+        shouldChangeToMobileUrl(request)?.let {
+            it.getMobileSite(url)?.let { newUrl ->
```

Comment:
My expectation here is that if I get a nonNull `MobileUrlReWriter.MobileSiteOnly` returned by `shouldChangeToMobileUrl`, calling`getMobileSite` will never be null because a valid Uri was already part of the logic. Does that make sense?

if not I see a bit weird that `mobileSiteOnlyForUri` returns `MobileSiteOnly?` if I can't use it.

Reply:
I gave this another go. The change is basically returning the newUrl directly in `shouldChangeToMobileUrl` and move `getMobileSite` to be a private extension function so it cannot be called elsewhere but within the `MobileUrlReWriter` class.